roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/StreetSection.java

```diff
@@ -198,16 +198,19 @@ public boolean isEnoughSpace(double length) {
     public void moveFirstCarToNextSection() {
         ICar firstCar = removeFirstCar();
         if (firstCar != null) {
-            if (firstCar.getCurrentSection() != firstCar.getDestination()) {
+            if (!Objects.equals(firstCar.getCurrentSection(), firstCar.getDestination())) {
                 IStreetSection nextSection = firstCar.getNextSection();
-                // Move physically first car to next section.
-                nextSection.addCar(firstCar);
-                // Move logically first car to next section.
-                firstCar.traverseToNextSection();
+                if (nextSection != null) {
+                    // Move physically first car to next section.
+                    nextSection.addCar(firstCar);
+                    // Move logically first car to next section.
+                    firstCar.traverseToNextSection();
+                }
```

Comment:
would an exception be useful here, to recognize if a car is lost?

Reply:
I’m not sure about this, as this scenario should never be reached.
But if the “if” case is handled there, so should be the else.